### PR TITLE
Create udes-ui-core package, add Button component and stories

### PR DESCRIPTION
**Pull request by squah-u**

### CODE REVIEW

File: packages/udes-ui-core/package.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@udes-ui/core",
+  "version": "0.0.1",
+  "description": "UDES UI Core",
+  "author": "Richard Whitefoot <mouseholdonline@gmail.com>",
```
Comment (richardwhitefoot) [inline packages/udes-ui-core/package.json]: Author should change

File: packages/udes-ui-core/package.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  "keywords": [
+    "udes",
+    "udes-ui"
+  ],
+  "dependencies": {
+    "@material-ui/core": "^4.7.0",
+    "@material-ui/styles": "^4.6.0",
+    "@types/react": "16.9.13",
+    "@types/styled-components": "^4.4.0",
+    "styled-components": "^4.4.1"
```
Comment (richardwhitefoot) [inline packages/udes-ui-core/package.json]: Renovate requires us to pin npm packages to exact versions

Comment (squah-u) [reply packages/udes-ui-core/package.json]: Thanks - fixed

Comment (squah-u) [reply packages/udes-ui-core/package.json]: Thanks, fixed.

File: packages/udes-ui-core/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## @udes-ui/core
+
+Core UDES React components
```
Comment (richardwhitefoot) [inline packages/udes-ui-core/README.md]: We should include developers docs on how to use these components

File: packages/udes-ui-storybook/storybook/stories/udes-ui-core/index.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import Stories from "./stories";
+import { ThemeProvider, theme } from "../../../../udes-ui-core/src";
+
+export default {
+  // eslint-disable-next-line react/display-name, react/prop-types
+  decorator: (storyFn) => (
+    <ThemeProvider theme={theme}>{storyFn()}</ThemeProvider>
```
Comment (richardwhitefoot) [inline packages/udes-ui-storybook/storybook/stories/udes-ui-core/index.jsx]: Wrapping the output in this decorator is a nice touch, however a side-effect is that the Prop Types for the `<Button>` components will not be generated under the 'Show info' button in Storybook - could you see if there is a workaround please.

File: packages/udes-ui-core/README.md (hunk 3 above)
Comment (richardwhitefoot) [reply packages/udes-ui-core/README.md]: I should note, the package README.md file is published on npmjs.com

File: packages/udes-ui-storybook/storybook/stories/udes-ui-core/index.jsx (hunk 4 above)
Comment (squah-u) [reply packages/udes-ui-storybook/storybook/stories/udes-ui-core/index.jsx]: Fixed by tweaking some webpack loader settings

**Comment by squah-u** [comment]
Fixed a README.md change that was in the wrong commit.

File: packages/udes-ui-core/src/theme.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  const merged = Object.assign({}, a, b);
+
+  for (const k in merged) {
+    if (
+      typeof a[k] == "object" &&
+      typeof b[k] == "object" &&
+      !Array.isArray(a[k]) && // Avoid merging arrays
+      !Array.isArray(b[k])
+    ) {
+      merged[k] = merge(a[k], b[k]);
```
[the remark this answers is not in the document]
Comment (squah-u) [reply packages/udes-ui-core/src/theme.ts]: Keys won't necessarily be present in both theme fragments. If a key is only present in one of the themes, the shallow merge done above by `Object.assign` will make sure it's in `merged`. If they're in both and the value is `{}`-like, then this line does a recursive merge.

**Pull request merged**